dpnp/dpnp_array.py

```diff
@@ -1252,12 +1252,48 @@ def searchsorted(self, v, side="left", sorter=None):
     @property
     def shape(self):
         """
-        Lengths of axes. A tuple of numbers represents size of each dimension.
+        Tuple of array dimensions.
 
-        Setter of this property involves reshaping without copy. If the array
-        cannot be reshaped without copy, it raises an exception.
+        The shape property is usually used to get the current shape of an array,
+        but may also be used to reshape the array in-place by assigning a tuple
+        of array dimensions to it. Unlike :obj:`dpnp.reshape`, only non-negative
+        values are supported to be set as new shape. Reshaping an array in-place
+        will fail if a copy is required.
 
-        .. seealso: :attr:`numpy.ndarray.shape`
+        For full documentation refer to :obj:`numpy.ndarray.shape`.
+
+        Note
+        ----
+        Using :obj:`dpnp.ndarray.reshape` or :obj:`dpnp.reshape is the
```

Comment:
Do you mean both of them or just `reshape`, kind of similar to what `NumPy` warns?
```suggestion
        Using :obj:`dpnp.ndarray.reshape` is the
```

Reply:
I mean to use `dpnp.rehsape(a, ...)` or `a.reshape(...)` is a preferable, because `shape.setter` is quite limited if it is need to change the shape.
IN general, yes, kind of similar with NumPy warning.

Reply:
I'll update the link to :obj:`dpnp.reshape`